resources/assets/js/components/ApplicantProfile/Skills/SkillSearchResults.tsx

```diff
@@ -0,0 +1,199 @@
+import * as React from "react";
+import { defineMessages, useIntl } from "react-intl";
+import { h2ComponentAccordionAddTriggerEvent } from "@hydrogen-design-system/system/dist/import/latest/components/accordion/scripts/accordion";
+import { Skill } from "../../../models/types";
+import { getLocale, localizeFieldNonNull } from "../../../helpers/localize";
+
+const resultMessages = defineMessages({
+  resultTitle: {
+    id: "results.title",
+    defaultMessage: "Results",
+    description: "The message displayed before the number of results returned.",
+  },
+  resultEmpty: {
+    id: "results.empty",
+    defaultMessage: "Nothing found",
+    description: "The message displayed if no results returned.",
+  },
+  resultAccordionExpand: {
+    id: "results.accordion.expand",
+    defaultMessage: "Click to view...",
+    description: "The message displayed to expand the accordion.",
+  },
+  resultAccordionAdd: {
+    id: "results.accordion.add",
+    defaultMessage: "Add",
+    description: "The label for the add button.",
+  },
+  resultAccordionAdded: {
+    id: "results.accordion.added",
+    defaultMessage: "Added!",
+    description: "The message displayed if skill added.",
+  },
+});
+
+interface SkillSearchResultsProps {
+  status: string;
+  results: Array<Skill> | null;
+  handleAddSkill: (skillId: number) => Promise<Skill>;
+}
+
+interface SkillSearchResultItemProps {
+  item: {
+    id: number;
+    name: { en: string; fr: string };
+    description: { en: string; fr: string };
+    isChecked?: boolean;
+  };
+  handleAddSkill: (skillId: number) => Promise<Skill>;
+}
+
+const SkillSearchResultItem: React.FunctionComponent<SkillSearchResultItemProps> = ({
+  item,
+  handleAddSkill,
+}: SkillSearchResultItemProps): React.ReactElement => {
+  const intl = useIntl();
+  const locale = getLocale(intl.locale);
+  const [added, setAdded] = React.useState(false);
```

Comment:
I think it would be worth leaving a comment here explaining why we have an `added` state in addition to `isChecked` coming from the prop - the fact that we want to show different UI if we've just recently added it compared to sometime in the past, and this helps distinguish that.